src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/MetadataTemplateServiceImpl.java

```diff
@@ -234,15 +227,46 @@ public MetadataRestriction setMetadataRestriction(Project project, MetadataTempl
 		return metadataRestrictionRepository.save(metadataRestrictionForFieldAndProject);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	@PreAuthorize("hasPermission(#template, 'canReadMetadataTemplate')")
+	public List<MetadataTemplateField> getPermittedFieldsForTemplate(MetadataTemplate template) {
+		List<MetadataTemplateField> fieldsForTemplate = fieldRepository.getMetadataFieldsForTemplate(template);
+
+		List<MetadataTemplateField> permittedFieldsForCurrentUser = getPermittedFieldsForCurrentUser(
+				template.getProject(), true);
+
+		return fieldsForTemplate.stream()
+				.filter(f -> permittedFieldsForCurrentUser.contains(f))
+				.collect(Collectors.toList());
+	}
+
 	/**
 	 * {@inheritDoc}
 	 */
 	@PreAuthorize("hasPermission(#project, 'isProjectOwner')")
 	@Override
-	public List<MetadataTemplateField> getPermittedFieldsForRole(Project project, ProjectRole role) {
+	public List<MetadataTemplateField> getPermittedFieldsForRole(Project project, ProjectRole role,
+			boolean includeTemplateFields) {
 		//get all fields for the project
 		List<MetadataTemplateField> metadataFieldsForProject = fieldRepository.getMetadataFieldsForProject(project);
 
+		if (includeTemplateFields) {
```

Comment:
Added a flag here for whether to include template fields when reading permitted fields for a user.